yt_idefix/data_structures.py

```diff
@@ -724,3 +793,277 @@ def _validate_units_override_keys(cls, units_override):
+    _version_regexp = re.compile(r"\d+\.\d+\.?\d*[-\w+]*")
+    _required_header_keyword = "PLUTOXdmf"
+    _default_definitions_header = "definitions.h"
+    _default_inifile = "pluto.ini"
+    _field_offset_index = None
+
+    def _parse_parameter_file(self):
+        # IdefixDataset._parse_parameter_file()
+        grid_file = os.path.join(
+            os.path.dirname(self.parameter_filename), "grid.out"
+        )  # data-loc/grid.out
+        (
+            self.parameter_filename[:-2] + "xmf"
+        )  # data.%04d.<dbl/flt>.h5 -> data.%04d.<dbl/flt>.xmf
+        out_file = os.path.join(
+            os.path.dirname(self.parameter_filename),
+            self.parameter_filename[-6:]
+            + ".out",  # data.%04d.<dbl/flt>.h5 -> <dbl/flt>.h5.out
+        )
+        self._default_inifile = os.path.join(
+            os.path.dirname(self.parameter_filename), self._default_inifile
+        )
+        self._default_definitions_header = os.path.join(
+            os.path.dirname(self.parameter_filename), self._default_definitions_header
+        )
+        if os.path.exists(self._default_definitions_header):
+            with open(self._default_definitions_header) as deftxt:
+                lines = deftxt.readlines()
+                for line in lines:
+                    if "USER_DEF_PARAMETERS" in line:
+                        self.inputParamCount = int(line.split()[-1])
+        with open(grid_file) as gridtxt:
+            txt = gridtxt.readlines()
+            for line in txt:
+                if "# DIMENSIONS" in line:
+                    self.dimensionality = int(line[-2])
+                    break
+            domain_left_edge = np.zeros(self.dimensionality, dtype=np.float64)
+            domain_right_edge = np.zeros(self.dimensionality, dtype=np.float64)
+            domain_dimensions = np.zeros(self.dimensionality, dtype=np.int64)
+            geom_str = None
+            count = 0
+            for line in txt:
+                if ("# X1" in line) or ("# X2" in line) or ("# X3" in line):
+                    tmp = (
+                        line.replace(",", "").replace("[", "").replace("]", "").split()
+                    )
+                    domain_left_edge[count] = float(tmp[2])
+                    domain_right_edge[count] = float(tmp[3])
+                    domain_dimensions[count] = int(tmp[4])
+                    count += 1
+                if "# GEOMETRY" in line:
+                    geom_str = (line.split()[-1]).lower()
+            for i in range(
+                count, self.dimensionality
+            ):  # These are dummy, may need some fix
+                domain_left_edge[i] = 0.0
+                domain_right_edge[i] = 1.0
+                domain_dimensions[i] = 1
+            self.domain_left_edge = domain_left_edge
+            self.domain_right_edge = domain_right_edge
+            self.domain_dimensions = domain_dimensions
+
+            self.geometry = self._parse_geometry(geom_str)
```

Comment:
I'm wondering why you have to explicitly assign `self.geometry` here, the parent class PlutoVtkDataset will take care of it.

Reply:
Agreed. But the parent class needs to be explicitly called

Reply:
I wanted to use the minimum number of files as possible. Since, ```grid.out``` has a lot of information on the grid, I thought using ```grid.out``` for parsing would exhaustively cover almost everything needed by yt. Let me know if this is a good idea and is worth implementing in the parent classes as well.

Reply:
> Agreed. But the parent class needs to be explicitly called

Yes, right now such an explicit call would most likely generate errors.

Reply:
> I wanted to use the minimum number of files as possible.

I'm totally agreed! I just gave you some suggestion there. However that means you need to do a lot of work and much of existing work will be dropped. Please think about that.